### PR TITLE
Fix building with GCC on Linux

### PR DESCRIPTION
**Pull request by q4a**
It's always good to be able to test game on another compiler (somtimes bugs happens in compiler)
In addition, I have not very pleasant experience with debugging with Clang on ARM Linux. GCC woks better in my case.
I'm not sure if we need add GCC to CI. It's more of a nice feature and it's okay if someone breaks it for a short time.

### CODE REVIEW

File: src/libs/math/include/math3d/color.h [hunk 1]
```diff
@@ -18,49 +18,14 @@
 class Color
 {
   public:
-    union {
```
Comment (Hammie) [inline src/libs/math/include/math3d/color.h:21]: What was the problem here?

Comment (q4a) [reply src/libs/math/include/math3d/color.h:21]: Current version gives error:
```
src/libs/math/include/math3d/color.h:38:32: error: member ‘Vector Color::<unnamed union>::<unnamed struct>::<unnamed union>::<unnamed union>::<unnamed struct>::c’ with constructor not allowed in anonymous aggregate
   38 |                         Vector c;
      |                                ^
```

GCC allows only:
https://gcc.gnu.org/onlinedocs/gcc/Unnamed-Fields.html
And GCC's implementation of the feature does not allow the unnamed struct to have members that have non-trivial constructors.

So this works in GCC:
```
union {
    struct
    {
        // Red
        float r;
        // Green
        float g;
        // Blue
        float b;
        // Transparency
        float a;
    };

    // Vector4 representation
    Vector4 v4;
};
```

And this is not:
```
union {
    struct
    {
        // rgb in vector
        Vector color;
        // Transparency
        float a;
    };

    // Vector4 representation
    Vector4 v4;
};
```

Comment (espkk) [reply src/libs/math/include/math3d/color.h:21]: That doesn't sound to be the case. Both classes have trivial(empty) constructors precisely for that reason.
Can you make a godbolt snippet?

Comment (q4a) [reply src/libs/math/include/math3d/color.h:21]: @espkk here you are: https://godbolt.org/z/8468cnoPs

Comment (espkk) [reply src/libs/math/include/math3d/color.h:21]: Move `default`ing into the class, that should work. I remember doing that, probably I missed it for Vector4

Comment (espkk) [reply src/libs/math/include/math3d/color.h:21]: IIRC, it's also possible to make a constructor inside the union to make default initialization explicit. I would avoid something like that, but the approach with `default` constructors should be totally fine and so far I don't see any non-standard or UB there

Comment (q4a) [reply src/libs/math/include/math3d/color.h:21]: @espkk ok. How to fix same error with non default constructor?
https://godbolt.org/z/caMfoEjTT

Comment (q4a) [reply src/libs/math/include/math3d/color.h:21]: @espkk I'm not sure how to do that. I tried `Vector color();` and `Vector color = new Vector();` - but it's not working: https://godbolt.org/z/aP5fqcjKd
Moving constructors into the class is not helping too for other constructors: https://godbolt.org/z/jaE3fz3dv

Comment (espkk) [reply src/libs/math/include/math3d/color.h:21]: Oh, gcc is so "special"
That one should work, but I'm not sure about UB https://godbolt.org/z/KavPdbf3r

Comment (q4a) [reply src/libs/math/include/math3d/color.h:21]: There is no unnamed structs in your case.
I can add names to structs and added it in some commits. Is it worth to add names in the `Color` class?

Comment (q4a) [reply src/libs/math/include/math3d/color.h:21]: https://godbolt.org/z/8dcWfGY9j - here is more complex/real example. If remove `gcc_will_fail_if_removed` - gcc will fail.

Comment (q4a) [reply src/libs/math/include/math3d/color.h:21]: @espkk Hi. Is there any news/problems to solve for this PR?

Comment (Hammie) [reply src/libs/math/include/math3d/color.h:21]: I'm happy with these changes

**Review by espkk — APPROVED** [review]

**Pull request merged**